common/persistence/sql/sqlExecutionManager.go

```diff
@@ -1325,23 +1455,39 @@ func (m *sqlExecutionManager) RangeCompleteTimerTask(request *p.RangeCompleteTim
 	return nil
 }
 
-// NewSQLMatchingPersistence creates an instance of ExecutionManager
+// NewSQLMatchingPersistence creates an instance of ExecutionStore
 func NewSQLMatchingPersistence(cfg config.SQL, logger bark.Logger) (p.ExecutionStore, error) {
-	var _, err = newConnection(cfg)
+	db, err := newConnection(cfg)
 	if err != nil {
 		return nil, err
 	}
-	return nil, nil
+	return &sqlExecutionManager{
+		sqlManager: sqlManager{
+			db:     db,
+			logger: logger,
+		},
+	}, nil
 }
 
+// getCurrentExecutionIfExists returns current execution or nil if none is found for the workflowID
+// locking it in the DB
 func getCurrentExecutionIfExists(tx *sqlx.Tx, shardID int64, domainID string, workflowID string) (*currentExecutionRow, error) {
```

Comment:
consider renaming to lockCurrentExecutionIfExist - this method no longer does a SELECT, it does a SELECT FOR UPDATE